### PR TITLE
Mehrzad / update client info

### PR DESCRIPTION
**Pull request by mehrzad-fs**

### CODE REVIEW

File: src/botPage/bot/TradeEngine/Balance.js [hunk 1]
```diff
@@ -1,25 +1,28 @@
-import { roundBalance } from '../../common/tools';
-import { info } from '../broadcast';
-import { observer as globalObserver } from '../../../common/utils/observer';
+import { roundBalance } from "../../common/tools";
```
Comment (sara-fs) [inline src/botPage/bot/TradeEngine/Balance.js:1]: Is there any reason to change it to double-quotation? eslint best practice is single-quot

File: src/botPage/bot/TradeEngine/Balance.js [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
+          return;
         }
-        // eslint-disable-next-line class-methods-use-this
-        getBalance(type) {
-            const balance = globalObserver.getState('balance');
-            const balance_str = `${balance}`;
-            return type === 'STR' ? balance_str : Number(balance);
+        if (data?.msg_type === "balance") {
+          const { currency } = data.balance;
+          const balance = roundBalance({ currency, balance: data.balance.balance });
```
Comment (sara-fs) [inline src/botPage/bot/TradeEngine/Balance.js:14]: what about this? 
const { currency, balance } = data.balance;
const data_balance = roundBalance({ currency, balance });

File: src/botPage/bot/TradeEngine/OpenContract.js [hunk 3]
```diff
@@ -1,101 +1,104 @@
-import { roundBalance } from '../../common/tools';
-import { doUntilDone } from '../tools';
-import { contractStatus, contractSettled, contract as broadcastContract } from '../broadcast';
-import { sell, openContractReceived } from './state/actions';
-import { observer } from '../../../common/utils/observer';
+import { roundBalance } from "../../common/tools";
```
Comment (sara-fs) [inline src/botPage/bot/TradeEngine/OpenContract.js:1]: same here

File: src/botPage/bot/TradeEngine/Proposal.js [hunk 4]
```diff
@@ -1,179 +1,177 @@
...
+      }
+
+      const to_buy = proposals.find(proposal => {
+        if (proposal.contract_type === contract_type && proposal.purchase_reference === this.getPurchaseReference()) {
+          // Below happens when a user has had one of the proposals return
+          // with a ContractBuyValidationError. We allow the logic to continue
+          // to here cause the opposite proposal may still be valid. Only once
+          // they attempt to purchase the errored proposal we will intervene.
+          if (proposal.error) {
+            const { error } = proposal.error.error;
```
Comment (sara-fs) [inline src/botPage/bot/TradeEngine/Proposal.js:35]: proposal.error.error or proposal.error ?

File: src/botPage/view/blockly/blocks/trade/components.js [hunk 5]
```diff
@@ -1,132 +1,127 @@
...
-        if (isLoggedInDeriv()) {
-            const activeAccount = getActiveAccount();
-            amountInput.appendField(activeAccount.currency, 'CURRENCY_LIST');
-        } else {
-            amountInput.appendField(config.lists.CURRENCY[0][0], 'CURRENCY_LIST');
-        }
+    if (block.type === "tradeOptions_payout") {
+      amountInput.appendField(`${translate("Payout")}:`);
+    } else {
+      amountInput.appendField(`${translate("Stake")}:     `);
```
Comment (sara-fs) [inline src/botPage/view/blockly/blocks/trade/components.js:72]: ` amountInput.appendField(`${translate("Stake")}: `);`

File: src/botPage/view/deriv/layout/Header/Header.jsx [hunk 6]
```diff
@@ -1,153 +1,166 @@
...
+          dispatch(setAccountSwiitcherLoader(true))  
+      });
+      syncWithDerivApp();
+    }
+  }, []);
 
-    const hideDropdown = e => !platformDropdownRef.current.contains(e.target) && setIsPlatformSwitcherOpen(false);
+  React.useEffect(() => {
+    dispatch(updateIsLooged(isLoggedIn()));
+  }, [is_logged]);
```
Comment (sara-fs) [inline src/botPage/view/deriv/layout/Header/Header.jsx:104]: when is_logged is changing why do we update the satate with isLoggedIn()?

File: src/botPage/bot/TradeEngine/OpenContract.js [hunk 7]
```diff
@@ -1,101 +1,104 @@
...
-                    }
-                }
-            });
+  class OpenContract extends Engine {
+    observeOpenContract() {
+      this.api.onMessage().subscribe(({ data }) => {
+        if (data?.error?.code) {
+          return;
         }
+        if (data?.msg_type === "proposal_open_contract") {
```
Comment (salarhfz-fs) [inline src/botPage/bot/TradeEngine/OpenContract.js:16]: Could we have an object including all the possible message types like an `enum`?

File: src/botPage/bot/TradeEngine/Balance.js (hunk 1 above)
Comment (mehrzad-fs) [reply src/botPage/bot/TradeEngine/Balance.js:1]: @sara-fs  I used single quat, It's automatically updated by eslint on creating PR,

File: src/botPage/bot/TradeEngine/OpenContract.js (hunk 7 above)
Comment (mehrzad-fs) [reply src/botPage/bot/TradeEngine/OpenContract.js:16]: Sure but it's out of this card scope we will update it

File: src/botPage/view/deriv/layout/Header/Header.jsx (hunk 6 above)
Comment (mehrzad-fs) [reply src/botPage/view/deriv/layout/Header/Header.jsx:104]: We can check it initially, but for the next step, we decide to handle login/logout and switch accounts without reloading. so we check if `is_logged` status changed

File: src/botPage/bot/TradeEngine/Proposal.js (hunk 4 above)
Comment (mehrzad-fs) [reply src/botPage/bot/TradeEngine/Proposal.js:35]: Recently, @sandeep-deriv updated the `binary-live-api` to `deriv-live-api`. the API response structure is different. for error we have and object like this 
error:{
   code: // error name
   error: // error content
}

**Pull request merged**